File: PZXVerificationCodeTextField/ViewController.m
//
//  ViewController.m
//  PZXVerificationCodeTextField
//
//  Created by pzx on 16/12/11.
//  Copyright © 2016年 pzx. All rights reserved.
//

#import "ViewController.h"
#import "PZXVerificationCodeView.h"
#import "PZXVerificationCodeTextField-Swift.h" // 自动生成的Swift桥接头文件

@interface ViewController () <PZXCodeInputViewDelegate>

@property(nonatomic,strong)PZXVerificationCodeView *cellView;

@property(nonatomic,strong)PZXVerificationCodeView *SpacingView;

@property(nonatomic,strong)PZXVerificationCodeView *grayView;


@property(nonatomic,strong)UITextField *TF;
- (IBAction)changeButtonPressed:(UIButton *)sender;
- (IBAction)outButtonPressed:(UIButton *)sender;

@property(nonatomic,strong)PZXVerificationCodeView *bottomLineInputView;
@property(nonatomic,strong)PZXCodeInputView *codeInputView;


@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.view.backgroundColor = [UIColor whiteColor];
    
    
    _cellView = [[PZXVerificationCodeView alloc]initWithFrame:CGRectMake(0, 100, self.view.frame.size.width, 60)];
    _cellView.selectedColor = [UIColor blackColor];
    //    _pzxView.center = self.view.center;
    //    _pzxView.deselectColor = [UIColor cyanColor];
    _cellView.VerificationCodeNum = 6;
    //    _pzxView.isSecure = YES;//密文
    _cellView.Spacing = 0;//每个格子间距属性
    [self.view addSubview:_cellView];
    
    _SpacingView = [[PZXVerificationCodeView alloc]initWithFrame:CGRectMake(0, 200, self.view.frame.size.width, 60)];
    _SpacingView.selectedColor = [UIColor blueColor];
    _SpacingView.deselectColor = [UIColor grayColor];
    //隐藏键盘游标
    _SpacingView.isCursorHidden = true;
    _SpacingView.borderWidth = 2;
    _SpacingView.VerificationCodeNum = 6;
    //    _pzxView.isSecure = YES;//密文
    _SpacingView.Spacing = 6;//每个格子间距属性
    [self.view addSubview:_SpacingView];
    
    
    _grayView = [[PZXVerificationCodeView alloc]initWithFrame:CGRectMake(0, 300, self.view.frame.size.width, 60)];
    _grayView.selectedColor = [UIColor clearColor];
    _grayView.deselectColor = [UIColor clearColor];
    _grayView.textFieldBackgroundColor = [UIColor groupTableViewBackgroundColor];
    _grayView.borderWidth = 0;
    _grayView.cornerRadius = 6;
    _grayView.VerificationCodeNum = 6;
    //    _pzxView.isSecure = YES;//密文
    _grayView.Spacing = 6;//每个格子间距属性
    [self.view addSubview:_grayView];
    
    
    _bottomLineInputView = [[PZXVerificationCodeView alloc]initWithFrame:CGRectMake(16, 400, self.view.frame.size.width - 32, 60)];
    _bottomLineInputView.selectedColor = [UIColor blackColor];
    _bottomLineInputView.deselectColor = [UIColor lightGrayColor];

    _bottomLineInputView.VerificationCodeNum = 6;
    _bottomLineInputView.lineStyle = true;
    _bottomLineInputView.Spacing = 16;//每根线间距属性
    [self.view addSubview:_bottomLineInputView];
    
    
    _codeInputView = [[PZXCodeInputView alloc] initWithNumberOfFields:4];
    _codeInputView.translatesAutoresizingMaskIntoConstraints = NO;
    _codeInputView.style = PZXCodeInputViewStyleDash;
    _codeInputView.delegate = self;
    [self.view addSubview:_codeInputView];
    
    [NSLayoutConstraint activateConstraints:@[
        [_codeInputView.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
        [_codeInputView.topAnchor constraintEqualToAnchor:self.view.topAnchor constant:500],
        [_codeInputView.heightAnchor constraintEqualToConstant:50],
        [_codeInputView.widthAnchor constraintEqualToConstant:300]
    ]];

    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}


- (IBAction)changeButtonPressed:(UIButton *)sender {
    
    _cellView.isSecure = !_cellView.isSecure;
}

- (IBAction)outButtonPressed:(UIButton *)sender {
    
    UIAlertController  *alert = [UIAlertController alertControllerWithTitle:@"验证码" message:[NSString stringWithFormat:@"%@",_cellView.vertificationCode] preferredStyle:UIAlertControllerStyleAlert];
    UIAlertAction *ok = [UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        
    }];
    
    [alert addAction:ok];
    [self presentViewController:alert animated:YES completion:nil];
}

-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    
    for (UITextField *tf in _cellView.textFieldArray) {
        
        [tf resignFirstResponder];
    }
    for (UITextField *tf in _bottomLineInputView.textFieldArray) {
        
        [tf resignFirstResponder];
    }
    
    [_codeInputView setResignFirstResponder];
    //    [self.view endEditing:YES];
    
}

#pragma mark - PZXCodeInputViewDelegate

- (void)codeInputViewDidFinishInput:(PZXCodeInputView *)inputView code:(NSString *)code {
    NSLog(@"Code entered: %@", code);
    // 在这里处理输入完成后的逻辑
}
@end

File: PZXVerificationCodeTextFieldSwiftDemo/PZXVerificationCodeTextFieldSwiftDemo/ViewController.swift
import UIKit

class ViewController: UIViewController, PZXCodeInputViewDelegate {

    private var cellView: PZXVerificationCodeView!
    private var spacingView: PZXVerificationCodeView!
    private var grayView: PZXVerificationCodeView!
    private var bottomLineInputView: PZXVerificationCodeView!
    private var codeInputView: PZXCodeInputView!

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white

        // Initialize cellView
        cellView = PZXVerificationCodeView(frame: CGRect(x: 0, y: 100, width: view.frame.size.width, height: 60))
        cellView.selectedColor = .black
        cellView.verificationCodeNum = 6
        cellView.spacing = 0
        view.addSubview(cellView)

        // Initialize spacingView
        spacingView = PZXVerificationCodeView(frame: CGRect(x: 0, y: 200, width: view.frame.size.width, height: 60))
        spacingView.selectedColor = .blue
        spacingView.deselectColor = .gray
        spacingView.isCursorHidden = true
        spacingView.borderWidth = 2
        spacingView.verificationCodeNum = 6
        spacingView.spacing = 6
        spacingView.pzx_keyboardType = .decimalPad;
        view.addSubview(spacingView)

        // Initialize grayView
        grayView = PZXVerificationCodeView(frame: CGRect(x: 0, y: 300, width: view.frame.size.width, height: 60))
        grayView.selectedColor = .clear
        grayView.deselectColor = .clear
        grayView.textFieldBackgroundColor = .groupTableViewBackground
        grayView.borderWidth = 0
        grayView.cornerRadius = 6
        grayView.verificationCodeNum = 6
        grayView.spacing = 6
        view.addSubview(grayView)

        // Initialize bottomLineInputView
        bottomLineInputView = PZXVerificationCodeView(frame: CGRect(x: 16, y: 400, width: view.frame.size.width - 32, height: 60))
        bottomLineInputView.selectedColor = .black
        bottomLineInputView.deselectColor = .lightGray
        bottomLineInputView.verificationCodeNum = 6
        bottomLineInputView.lineStyle = true
        bottomLineInputView.spacing = 16
        view.addSubview(bottomLineInputView)

        // Initialize codeInputView
        codeInputView = PZXCodeInputView(numberOfFields: 4)
        codeInputView.translatesAutoresizingMaskIntoConstraints = false
        codeInputView.style = .dash
        codeInputView.delegate = self
        view.addSubview(codeInputView)

        NSLayoutConstraint.activate([
            codeInputView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
            codeInputView.topAnchor.constraint(equalTo: view.topAnchor, constant: 500),
            codeInputView.heightAnchor.constraint(equalToConstant: 50),
            codeInputView.widthAnchor.constraint(equalToConstant: 300)
        ])
        
        grayView.pzx_becomeFirstResponder(1)
    }

    @objc func changeButtonPressed(_ sender: UIButton) {
        cellView.isSecure.toggle()
    }

    @objc func outButtonPressed(_ sender: UIButton) {
        let alert = UIAlertController(title: "验证码", message: cellView.vertificationCode, preferredStyle: .alert)
        let okAction = UIAlertAction(title: "确定", style: .default, handler: nil)
        alert.addAction(okAction)
        present(alert, animated: true, completion: nil)
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
//        for textField as! UIText in cellView.textFieldArray {
//            textField.resignFirstResponder()
//        }
//        for textField in bottomLineInputView.textFieldArray {
//            textField.resignFirstResponder()
//        }
//        codeInputView.resignFirstResponder()
        self.view.endEditing(true)
    }

    // MARK: - PZXCodeInputViewDelegate
    func codeInputViewDidFinishInput(_ inputView: PZXCodeInputView, code: String) {
        print("Code entered: \(code)")
        // Handle logic after input is completed
    }
}
